src/VirtualFile/VirtualPath.cs

```diff
@@ -142,11 +143,11 @@ internal static VirtualPath CreateNonRelative(string value)
         return value;
     }
 
-    internal Stream OpenFile() => Files.GetFileInfo(Path).CreateReadStream();
+    internal Stream OpenFile(IFileProvider files = null) => (files ?? Files).GetFileInfo(Path).CreateReadStream();
```

Comment:
Do we need a default one here? I'd rather be explicit and just remove the Files property if we can

Reply:
No, we don't need it to default to null, works without it. I wasn't sure if there were plans for Files so tried to keep it backwards compatible.